tests/test_resources.py

```diff
@@ -469,3 +469,47 @@ def test_resources_invalid_definitions(hq_env: HqEnv):
         args=["submit", "--resource", "foo=0", "--", "sleep", "1"],
         expect_fail="Zero resources cannot be requested",
     )
+
+
+def test_resources_and_priorities_one_by_one_submit(hq_env: HqEnv):
+    hq_env.start_server()
+    hq_env.start_worker(args=["--resource", "foo=[1, 2, 3, 4]"], cpus=8)
+
+    for i in range(1, 5):
+        hq_env.command(["submit", "--resource", "foo=2", "--", "sleep", "100"])
+        if i < 3:
+            wait_for_job_state(hq_env, i, "RUNNING")
+    time.sleep(0.6)
+    wait_for_job_state(hq_env, [1, 2], "RUNNING")
+    wait_for_job_state(hq_env, [3, 4], "WAITING")
+    time.sleep(1.0)
+    hq_env.command(["submit", "--cpus=2", "--priority=-1", "--", "sleep", "100"])
+    table = list_jobs(hq_env)
+    assert table.get_column_value("State") == ["RUNNING", "RUNNING", "WAITING", "WAITING", "RUNNING"]
+
+
+def test_resources_and_priorities_submit_before_worker(hq_env: HqEnv):
+    hq_env.start_server()
+    for i in range(1, 5):
+        hq_env.command(["submit", "--resource", "foo=2", "--", "sleep", "100"])
+    hq_env.command(["submit", "--cpus=2", "--priority=-1", "--", "sleep", "100"])
+    hq_env.start_worker(args=["--resource", "foo=[1, 2, 3, 4]"], cpus=8)
+    time.sleep(2)
```

Comment:
Let's wait for the state of the last job to be running instead of sleeping here.

Reply:
Here I really want to check that after 2s the situation is in this particular state. It would be really weird if we cannot schedule few tasks in 2s.

Reply:
Ok. I will ping you if this fails on CI :laughing:

Reply:
Ok, used `wait_until` with 5s timeout. It is slightly more than I would like, but hopefully catches potential bug that low priority task is started with delay.